sdk/remoterendering/mixedreality-remoterendering/package.json

```
@@ -0,0 +1,140 @@
+{
+  "name": "@azure/mixedreality-remoterendering",
+  "version": "1.0.0-beta.1",
+  "description": "Example project for learning how to build a client library",
+  "sdk-type": "client",
+  "main": "dist/index.js",
+  "module": "dist-esm/src/index.js",
+  "browser": {
+    "./dist-esm/test/utils/env.js": "./dist-esm/test/utils/env.browser.js"
+  },
+  "types": "types/mixedreality-remoterendering.d.ts",
+  "//metadata": {
+    "constantPaths": [
+      {
+        "path": "src/generated/generatedClientContext.ts",
+        "prefix": "packageVersion"
+      },
+      {
+        "path": "src/constants.ts",
+        "prefix": "SDK_VERSION"
+      }
+    ]
+  },
+  "scripts": {
+    "audit": "node ../../../common/scripts/rush-audit.js && rimraf node_modules package-lock.json && npm i --package-lock-only 2>&1 && npm audit",
+    "build:browser": "tsc -p . && cross-env ONLY_BROWSER=true rollup -c 2>&1",
+    "build:node": "tsc -p . && cross-env ONLY_NODE=true rollup -c 2>&1",
+    "build:test": "tsc -p . && rollup -c 2>&1",
+    "build": "tsc -p . && rollup -c 2>&1 && api-extractor run --local",
+    "check-format": "prettier --list-different --config ../../../.prettierrc.json --ignore-path ../../../.prettierignore \"src/**/*.ts\" \"test/**/*.ts\" \"samples-dev/**/*.ts\" \"*.{js,json}\"",
+    "clean": "rimraf dist dist-* test-dist temp types *.tgz *.log",
+    "docs": "typedoc --excludePrivate --excludeNotExported --excludeExternals --stripInternal --mode file --out ./dist/docs ./src",
+    "build:samples": "echo Obsolete.",
+    "execute:samples": "dev-tool samples run samples-dev",
+    "extract-api": "tsc -p . && api-extractor run --local",
+    "format": "prettier --write --config ../../../.prettierrc.json --ignore-path ../../../.prettierignore \"src/**/*.ts\" \"test/**/*.ts\" \"samples-dev/**/*.ts\" \"*.{js,json}\"",
+    "generate:client": "autorest --typescript --v3 swagger",
+    "integration-test:browser": "karma start --single-run",
+    "integration-test:node": "nyc mocha -r esm --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 5000000 --full-trace \"dist-esm/test/{,!(browser)/**/}/*.spec.js\"",
+    "integration-test": "npm run integration-test:node && npm run integration-test:browser",
+    "lint:fix": "eslint package.json api-extractor.json src test --ext .ts --fix --fix-type [problem,suggestion]",
+    "lint": "eslint package.json tsconfig.json api-extractor.json src test --ext .ts -f html -o mixedreality-remoterendering-lintReport.html || exit 0",
```

I would rather make sure this PR is lint errors free so errors do not pile up later. Also, no need to lint `tsconfig.json`.
```suggestion
    "lint": "eslint package.json api-extractor.json src test --ext .ts",
```

---

It turns out that many of the SDKs use "exit 0" to get their pipelines green (I discovered that after strugging to make the linter happy earlier in development). If I turn this on, it will fail because we switched to node >= 12 (among other things).

I will do a linting pass and fix most of the issues. (Btw: The commenting style of generated files is inconsistent with the linter.)

I will also ignore warnings where internal APIs take a class as a parameter. IMO: adding interfaces to match those classes is more likely to introduce bugs than resolve them, especially when the class is generated, so I cannot force a match between the class and the interface.

---

(Actually I could possibly introduce an artificial parent with a autorest customization, but I feel that is overkill.)

---

I'll resolve this for now, but if you can recommend a viable strategy, please reopen.

---

Here are a few thoughts:
- we're ignoring linting issues in some packages where we could not fix all linting issues back at the time we added lint checking but all new packages should be lint-error free.
- for comments, we follow this syntax: https://tsdoc.org/pages/tags/param/. Most of issues will be fixed by adding a dash between the param name and the description.
- for the no class as param issue, you can ignore it similar to https://github.com/Azure/azure-sdk-for-js/blob/e823b26f4600a698eb1c06f48da9b780fb2331a7/sdk/search/search-documents/test/public/utils/setup.ts#L231

---

Also, the node version thing is being updated in this PR: https://github.com/Azure/azure-sdk-for-js/pull/15996.

---

The PR is now all clean except for the node version. It looks like the fix is in active development. Is there an ETA?

---

cc @ramya-rao-a I believe you would want to get your PR ready before the releases, right? so perhaps we can get it merged today.

---

@MalcolmTyrrell the PR has been merged. Please merge latest main into yours.

---

Linter runs clean now. Thanks!